ISE 500 when non-UUID Karton identifier is added

**Environment information**
* MWDB version (from `/about`):
* Installation method:
  - [ ] mwdb.cert.pl service
  - [ ] From PyPi (`pip install mwdb-core`)
  - [x] From docker-compose
  - [ ] Other (please explain)
* Plugins installed:

**Behaviour the bug (what happened?)**

<!-- What actually happened? -->
```
file.add_metakey("karton", "xd")
---------------------------------------------------------------------------
HTTPError                                 Traceback (most recent call last)
/usr/local/lib/python3.8/dist-packages/mwdblib/api.py in perform_request(self, method, url, *args, **kwargs)
    107             response = self.session.request(method, url, *args, **kwargs)
--> 108             response.raise_for_status()
    109             return response

~/mwdb-core/venv/lib/python3.8/site-packages/requests/models.py in raise_for_status(self)
    952         if http_error_msg:
--> 953             raise HTTPError(http_error_msg, response=self)
    954 

HTTPError: 500 Server Error: INTERNAL SERVER ERROR for url: http://127.0.0.1/api/object/a6c813c82281dce9ff4508fad03b00db3e910aaa19fd18a6b286570bb879fd88/meta
```

```
mwdb_1      | [ERROR] uWSGIWorker2Core0 - service.error_router:72 - Unhandled exception occurred
mwdb_1      | Traceback (most recent call last):
mwdb_1      |   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1950, in full_dispatch_request
mwdb_1      |     rv = self.dispatch_request()
mwdb_1      |   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1936, in dispatch_request
mwdb_1      |     return self.view_functions[rule.endpoint](**req.view_args)
mwdb_1      |   File "/usr/local/lib/python3.6/site-packages/flask_restful/__init__.py", line 480, in wrapper
mwdb_1      |     resp = resource(*args, **kwargs)
mwdb_1      |   File "/usr/local/lib/python3.6/site-packages/flask/views.py", line 89, in view
mwdb_1      |     return self.dispatch_request(*args, **kwargs)
mwdb_1      |   File "/usr/local/lib/python3.6/site-packages/flask_restful/__init__.py", line 595, in dispatch_request
mwdb_1      |     resp = meth(*args, **kwargs)
mwdb_1      |   File "./mwdb/resources/__init__.py", line 46, in endpoint
mwdb_1      |     return f(*args, **kwargs)
mwdb_1      |   File "./mwdb/resources/metakey.py", line 152, in post
mwdb_1      |     is_new = db_object.add_attribute(key, value)
mwdb_1      |   File "./mwdb/model/object.py", line 758, in add_attribute
mwdb_1      |     karton_id = UUID(value)
mwdb_1      |   File "/usr/local/lib/python3.6/uuid.py", line 140, in __init__
mwdb_1      |     raise ValueError('badly formed hexadecimal UUID string')
mwdb_1      | ValueError: badly formed hexadecimal UUID string
```

**Expected behaviour**

<!-- What did you expect to happen? -->
- HTTP 400 with validation error

**Additional context**

<!-- Add any other context about the problem here. -->
- Applies to legacy `/meta` behavior, we need to check if it applies to new Karton endpoints as well.

--- a/mwdb/resources/__init__.py
+++ b/mwdb/resources/__init__.py
@@ -1,3 +1,4 @@
+import uuid
 from functools import wraps
 from json import JSONDecodeError
 
@@ -128,3 +129,11 @@ def get_shares_for_upload(upload_as):
         share_with = [share_group, Group.get_by_name(g.auth_user.login)]
 
     return share_with
+
+
+def is_valid_uuid(value):
+    try:
+        uuid.UUID(value)
+    except ValueError:
+        return False
+    return True

--- a/mwdb/resources/karton.py
+++ b/mwdb/resources/karton.py
@@ -1,6 +1,6 @@
 from flask import request
 from flask_restful import Resource
-from werkzeug.exceptions import NotFound
+from werkzeug.exceptions import BadRequest, NotFound
 
 from mwdb.core.capabilities import Capabilities
 from mwdb.model import KartonAnalysis, Object
@@ -10,7 +10,13 @@ from mwdb.schema.karton import (
     KartonSubmitAnalysisRequestSchema,
 )
 
-from . import access_object, loads_schema, requires_authorization, requires_capabilities
+from . import (
+    access_object,
+    is_valid_uuid,
+    loads_schema,
+    requires_authorization,
+    requires_capabilities,
+)
 
 
 class KartonObjectResource(Resource):
@@ -211,6 +217,8 @@ class KartonAnalysisResource(Resource):
                 content:
                   application/json:
                     schema: KartonItemResponseSchema
+            400:
+                description: When analysis_id is not UUID value
             403:
                 description: When user doesn't have `karton_assign` capability.
             404:
@@ -221,6 +229,9 @@ class KartonAnalysisResource(Resource):
         db_object = access_object(type, identifier)
         if db_object is None:
             raise NotFound("Object not found")
+
+        if not is_valid_uuid(analysis_id):
+            raise BadRequest("ValidationError: analysis_id must be UUID value")
 
         analysis, _ = db_object.assign_analysis(analysis_id)
         schema = KartonItemResponseSchema()

--- a/mwdb/resources/karton.py
+++ b/mwdb/resources/karton.py
@@ -231,7 +231,7 @@ class KartonAnalysisResource(Resource):
             raise NotFound("Object not found")
 
         if not is_valid_uuid(analysis_id):
-            raise BadRequest("ValidationError: analysis_id must be UUID value")
+            raise BadRequest("'karton' attribute accepts only UUID values")
 
         analysis, _ = db_object.assign_analysis(analysis_id)
         schema = KartonItemResponseSchema()

--- a/mwdb/resources/metakey.py
+++ b/mwdb/resources/metakey.py
@@ -23,6 +23,7 @@ from mwdb.schema.metakey import (
 
 from . import (
     access_object,
+    is_valid_uuid,
     load_schema,
     loads_schema,
     logger,
@@ -132,6 +133,8 @@ class MetakeyResource(Resource):
                 content:
                   application/json:
                     schema: MetakeyListResponseSchema
+            400:
+                description: For karton attribute when value is not UUID
             404:
                 description: |
                     When object doesn't exist or user doesn't have
@@ -149,6 +152,10 @@ class MetakeyResource(Resource):
 
         key = obj["key"]
         value = obj["value"]
+
+        if key == "karton" and not is_valid_uuid(value):
+            raise BadRequest("ValidationError: for karton attribute value must be UUID")
+
         is_new = db_object.add_attribute(key, value)
         if is_new is None:
             raise NotFound(

--- a/mwdb/resources/metakey.py
+++ b/mwdb/resources/metakey.py
@@ -154,7 +154,7 @@ class MetakeyResource(Resource):
         value = obj["value"]
 
         if key == "karton" and not is_valid_uuid(value):
-            raise BadRequest("ValidationError: for karton attribute value must be UUID")
+            raise BadRequest("'karton' attribute accepts only UUID values")
 
         is_new = db_object.add_attribute(key, value)
         if is_new is None:

